File: token-validation-jaxrs/src/test/java/no/nav/security/token/support/jaxrs/rest/TokenResource.java
package no.nav.security.token.support.jaxrs.rest;

import jakarta.ws.rs.GET;
import jakarta.ws.rs.Path;
import jakarta.ws.rs.core.Response;
import no.nav.security.token.support.core.api.Unprotected;
import no.nav.security.token.support.jaxrs.JaxrsTokenValidationContextHolder;

@Path("echo")
@Unprotected
public class TokenResource {

    @GET
    @Path("token")
    public Response getToken() {
        return Response.ok()
                .entity(JaxrsTokenValidationContextHolder.getHolder().getTokenValidationContext().getJwtToken("protected").getEncodedToken())
                .build();
    }
}

File: token-validation-jaxrs/src/test/kotlin/no/nav/security/token/support/jaxrs/rest/TokenResource.kt
package no.nav.security.token.support.jaxrs.rest

import jakarta.ws.rs.GET
import jakarta.ws.rs.Path
import jakarta.ws.rs.core.Response.ok
import no.nav.security.token.support.core.api.Unprotected
import no.nav.security.token.support.jaxrs.JaxrsTokenValidationContextHolder.getHolder

@Path("echo")
@Unprotected
class TokenResource {

    @get:Path("token")
    @get:GET
    val token = ok()
            .entity(getHolder().getTokenValidationContext().getJwtToken("protected")!!.encodedToken)
            .build()
}